### PR TITLE
fix (glog script): remove invalid param from sed

### LINKED ISSUE #33966: Failure when installing glog on M1 Mac
### Description

After running `pod install` in a new project, the command failed to execute `sed -i '' 's/\@ac_cv_have_libgflags\@/0/' src/glog/logging.h.in` due to what seems to be incorrect syntax (there is an empty string after the `-i` flag, causing sed to interpret the actual replace string as a file path). Is there something obscure I'm missing here, or is this a genuine typo?

### Version

7.0.3

### Output of `npx react-native info`

System:
    OS: macOS 12.4
    CPU: (10) arm64 Apple M1 Pro
    Memory: 157.81 MB / 16.00 GB
    Shell: 3.4.1 - /opt/homebrew/bin/fish
  Binaries:
    Node: 18.2.0 - /opt/homebrew/bin/node
    Yarn: Not Found
    npm: 8.9.0 - /opt/homebrew/bin/npm
    Watchman: 2022.03.21.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.11.0 - /nix/store/48ikzg0v85glw6jcwfdx1y5w0dk7zj14-cocoapods-1.11.0/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK: Not Found
  IDEs:
    Android Studio: 2021.2 AI-212.5712.43.2112.8512546
    Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.15 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: Not Found
    react-native: Not Found
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

```bash
npx react-native init AwesomeProject --template react-native-template-typescript

# at this point it fails midway, and suggests you cd into the `ios` dir and run `pod install`

cd AwesomeProject/ios

pod install
```

### Snack, code example, screenshot, or link to a repository

```
patching file config.sub
sed: can't read s/\@ac_cv_have_libgflags\@/0/: No such file or directory
```

### PR DESCRIPTION
**Pull request by alphashuro**
## Summary

This is what I had to do locally to fix #33966 

## Changelog

[iOS] [Fixed] - Fix sed error when installing `glog`

## Test Plan

After this, `pod install` installed `glog` successfully 

image

### CODE REVIEW

File: scripts/ios-configure-glog.sh [hunk 1]
```diff
@@ -50,8 +50,8 @@ if [ -h "test-driver" ]; then
 fi
 
 # Manually disable gflags include to fix issue https://github.com/facebook/react-native/issues/28446
-sed -i '' 's/\@ac_cv_have_libgflags\@/0/' src/glog/logging.h.in
-sed -i '' 's/HAVE_LIB_GFLAGS/HAVE_LIB_GFLAGS_DISABLED/' src/config.h.in
```
Comment (cortinico) [inline scripts/ios-configure-glog.sh:54]: This is needed as you're not using Mac OS `sed` but you replaced it with GNU `sed`.
If we merge this, I have the suspect this is going to break for the majority of OSX users: https://unix.stackexchange.com/a/131940

Comment (alphashuro) [reply scripts/ios-configure-glog.sh:54]: Ah I see! In that case, let me change it slightly to support both instead, one moment...

File: scripts/ios-configure-glog.sh [hunk 2]
```diff
@@ -50,8 +50,8 @@ if [ -h "test-driver" ]; then
...
+sed -i 's/\@ac_cv_have_libgflags\@/0/' src/glog/logging.h.in
+sed -i 's/HAVE_LIB_GFLAGS/HAVE_LIB_GFLAGS_DISABLED/' src/config.h.in
```
Comment (alphashuro) [inline scripts/ios-configure-glog.sh:54]:
```suggestion
sed -i.bak -e 's/\@ac_cv_have_libgflags\@/0/' src/glog/logging.h.in && rm src/glog/logging.h.in.bak
sed -i.bak -e 's/HAVE_LIB_GFLAGS/HAVE_LIB_GFLAGS_DISABLED/' src/config.h.in && rm src/config.h.in.bak
```

File: scripts/ios-configure-glog.sh (hunk 1 above)
Comment (alphashuro) [reply scripts/ios-configure-glog.sh:54]: Updated! please let me know what you think

Comment (cortinico) [reply scripts/ios-configure-glog.sh:54]: Could you provide some refence on where the `-i.bak` is defined?

Comment (alphashuro) [reply scripts/ios-configure-glog.sh:54]: In both the [gnu sed](https://linux.die.net/man/1/sed) and [macOS sed](https://www.unix.com/man-page/osx/1/sed/) `-i` takes an extension as an optional parameter for creating a backup file, however if it is not passed in it reads '-e' as the extension instead. `.bak` in particular is an arbitrary choice but makes the backup extension explicit especially for the subsequent deletion.

Comment (alphashuro) [reply scripts/ios-configure-glog.sh:54]: @cortinico is this what you were looking for? or did you mean something different?

Comment (cortinico) [reply scripts/ios-configure-glog.sh:54]: Yup thanks for clarifying 👍

**Review by cortinico — APPROVED** [review]

File: scripts/ios-configure-glog.sh (hunk 1 above)
Comment (alphashuro) [reply scripts/ios-configure-glog.sh:54]: @cortinico I see that the PR failed lint tests, but I can't access the tests to see what's wrong. If you have access to these would you mind helping me find out what's causing it to fail?

Comment (charlesbdudley) [reply scripts/ios-configure-glog.sh:54]: @alphashuro I'm taking care of this right now. The tab literal characters in the patch content is making our internal linters angry. Stand by :)

**Pull request closed without merging**